app/lib/import_export.rb

```diff
@@ -148,38 +149,25 @@ def update_headers(data)
   # Finds the symptoms needed for the reports columns
   def update_assessment_symptom_data(data, patients)
     # Don't update if assessment symptom data isn't needed
-    return unless data.dig(:assessments, :checked)&.include?(:symptoms)
+    return [] unless data.dig(:assessments, :checked)&.include?(:symptoms)
 
     data[:assessments][:checked].delete(:symptoms)
     data[:assessments][:headers].delete('Symptoms Reported')
 
-    # For large queries (e.g. 100k patients) of the `Symptom.where(...) query below
-    # we saw MySQL give up on queries and throw an exception.
-    # The workaround for this is to collect the symptom ids in a set and find them
-    # using batches of patients.
-    symptoms = Set.new
-    patients.pluck(:id).in_groups_of(10_000, false) do |batch|
-      Symptom.where(
-        condition_id: ReportedCondition.where(
-          assessment_id: Assessment.where(
-            patient_id: batch
-          ).pluck(:id)
-        ).pluck(:id)
-      ).where.not(label: nil).where.not(name: nil).distinct.pluck(:name, :label).each { |symptom| symptoms.add(symptom) }
-    end
-    symptom_names_and_labels = symptoms.to_a.sort { |a, b| a[1] <=> b[1] }.transpose
+    # Query symptom names and labels
+    threshold_condition_hashes = ReportedCondition.where(assessment_id: Assessment.where(patient_id: patients)).distinct.select(:threshold_condition_hash)
+    symptom_names_and_labels = Symptom.where(condition_id: ThresholdCondition.where(threshold_condition_hash: threshold_condition_hashes))
+                                      .where.not(label: nil).where.not(name: nil).order(:label).distinct.pluck(:name, :label).transpose
```

Comment:
@timwongj were you able to write a benchmark script for this specific cahnge?

Reply:
similar to how we measured in this PR https://github.com/SaraAlert/SaraAlert/pull/917

Reply:
```
Benchmark.ips do |x|
    x.report('old_method') {
      symptoms = Set.new
      @patients.pluck(:id).in_groups_of(10_000, false) do |batch|
        Symptom.where(
          condition_id: ReportedCondition.where(
            assessment_id: Assessment.where(
              patient_id: batch
            ).pluck(:id)
          ).pluck(:id)
        ).where.not(label: nil).where.not(name: nil).distinct.pluck(:name, :label).each { |symptom| symptoms.add(symptom) }
      end
      symptom_names_and_labels = symptoms.to_a.sort { |a, b| a[1] <=> b[1] }.transpose
    }

    x.report('new_method') {
      threshold_condition_hashes = ReportedCondition.where(assessment_id: Assessment.where(patient_id: @patients)).distinct.select(:threshold_condition_hash)
      symptom_names_and_labels = Symptom.where(condition_id: ThresholdCondition.where(threshold_condition_hash: threshold_condition_hashes))
                                        .where.not(label: nil).where.not(name: nil).order(:label).distinct.pluck(:name, :label).transpose
    }

    x.compare!
  end
```

For 100 patients:
```
Warming up --------------------------------------
          old_method     1.000  i/100ms
          new_method     2.000  i/100ms
Calculating -------------------------------------
          old_method      8.080  (±12.4%) i/s -     41.000  in   5.099661s
          new_method     26.091  (± 3.8%) i/s -    132.000  in   5.068351s

Comparison:
          new_method:       26.1 i/s
          old_method:        8.1 i/s - 3.23x  (± 0.00) slower
```

For 500 patients:
```
Warming up --------------------------------------
          old_method     1.000  i/100ms
          new_method     1.000  i/100ms
Calculating -------------------------------------
          old_method      1.696  (± 0.0%) i/s -      9.000  in   5.317040s
          new_method      5.741  (± 0.0%) i/s -     29.000  in   5.051961s

Comparison:
          new_method:        5.7 i/s
          old_method:        1.7 i/s - 3.38x  (± 0.00) slower
```

For 1000 patients:
```
Warming up --------------------------------------
          old_method     1.000  i/100ms
          new_method     1.000  i/100ms
Calculating -------------------------------------
          old_method      0.262  (± 0.0%) i/s -      2.000  in   7.633471s
          new_method      2.899  (± 0.0%) i/s -     15.000  in   5.340975s

Comparison:
          new_method:        2.9 i/s
          old_method:        0.3 i/s - 11.05x  (± 0.00) slower
```

For 5,000 patients:
```
Warming up --------------------------------------
          old_method     1.000  i/100ms
          new_method    20.000  i/100ms
Calculating -------------------------------------
          old_method      0.252  (± 0.0%) i/s -      2.000  in   7.945374s
          new_method    200.958  (±10.0%) i/s -      1.000k in   5.020154s

Comparison:
          new_method:      201.0 i/s
          old_method:        0.3 i/s - 798.34x  (± 0.00) slower
```

For 10,000 patients:
```
Warming up --------------------------------------
          old_method     1.000  i/100ms
          new_method    20.000  i/100ms
Calculating -------------------------------------
          old_method      0.183  (± 0.0%) i/s -      1.000  in   5.460444s
          new_method    196.162  (±11.2%) i/s -    980.000  in   5.091774s

Comparison:
          new_method:      196.2 i/s
          old_method:        0.2 i/s - 1071.13x  (± 0.00) slower
```

For 30,000 patients:
```
Warming up --------------------------------------
          old_method     1.000  i/100ms
          new_method    15.000  i/100ms
Calculating -------------------------------------
          old_method      0.080  (± 0.0%) i/s -      1.000  in  12.438493s
          new_method    186.633  (±11.8%) i/s -    930.000  in   5.065473s

Comparison:
          new_method:      186.6 i/s
          old_method:        0.1 i/s - 2321.44x  (± 0.00) slower
```

Reply:
The reason for this type of scaling is because the new method scales with the number of `threshold_conditions` which is based on the number of jurisdictions * number of updates via admin:import/update jur script instead of `reported_conditions` which scales with the number of patients/assessments/symptoms

Reply:
Just did the same and got this result on 500k
```
Comparison:
cur_threshold_method:        0.6 i/s
      new_set_method:        0.0 i/s - 52.63x  (± 0.00) slower
```